src/shared/sol-util-file.c

```diff
@@ -565,3 +566,23 @@ sol_util_move_file(const char *old_path, const char *new_path, mode_t mode)
 
     return r;
 }
+
+SOL_API bool
+sol_util_busy_wait_file(const char *path, uint64_t nanoseconds)
+{
+    struct stat st;
+    struct timespec start = sol_util_timespec_get_current();
+
+    while (stat(path, &st)) {
+        struct timespec elapsed, now = sol_util_timespec_get_current();
+
+        sol_util_timespec_sub(&now, &start, &elapsed);
+        if ((uint64_t)elapsed.tv_sec >= (nanoseconds / SOL_NSEC_PER_SEC) &&
+            (uint64_t)elapsed.tv_nsec > (nanoseconds % SOL_NSEC_PER_SEC))
+            return false;
+
+        sched_yield();
```

Comment:
On 24-03-2016 11:19, Iván Briano wrote:
> In src/shared/sol-util-file.c
> <https://github.com/solettaproject/soletta/pull/1715#discussion_r57322579>:
>
>> +
>> +SOL_API bool
>> +sol_util_busy_wait_file(const char *path, uint64_t nanoseconds)
>> +{
>> +    struct stat st;
>> +    struct timespec start = sol_util_timespec_get_current();
>> +
>> +    while (stat(path, &st)) {
>> +        struct timespec elapsed, now = sol_util_timespec_get_current();
>> +
>> +        sol_util_timespec_sub(&now, &start, &elapsed);
>> +        if ((uint64_t)elapsed.tv_sec >= (nanoseconds / SOL_NSEC_PER_SEC) &&
>> +            (uint64_t)elapsed.tv_nsec > (nanoseconds % SOL_NSEC_PER_SEC))
>> +            return false;
>> +
>> +        sched_yield();
>
> On Thu, 24 Mar 2016 07:14:54 -0700, Gustavo Lima Chaves wrote: > + >
> +SOL_API bool > +sol_util_busy_wait_file(const char *path, uint64_t
> nanoseconds) > +{ > + struct stat st; > + struct timespec start =
> sol_util_timespec_get_current(); > + > + while (stat(path, &st)) { > +
> struct timespec elapsed, now = sol_util_timespec_get_current(); > + > +
> sol_util_timespec_sub(&now, &start, &elapsed); > + if
> ((uint64_t)elapsed.tv_sec >= (nanoseconds / SOL_NSEC_PER_SEC) && > +
> (uint64_t)elapsed.tv_nsec > (nanoseconds % SOL_NSEC_PER_SEC)) > + return
> false; > + > + sched_yield(); This file is compiled under
> ``FEATURE_FILESYSTEM``. ``sched_yield()``, though, is Linux-only.
> Shouldn't we ifdef the function, then?
> The whole of IIO, as it exists right now, is Linux only.

That's true. But this change was made in sol-util-file.c. As I borrowed 
code from memmap storage, I put it on shared place.

>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/solettaproject/soletta/pull/1715/files/3959e5f4e882cf2d4cc813e4dc955d543e8648ff#r57322579>
>